pallets/dapp-staking-v3/src/lib.rs

```diff
@@ -1695,24 +1718,26 @@ pub mod pallet {
                 tier_id.saturating_inc();
             }
 
+            // TODO: what if multiple dApps satisfy the tier entry threshold but there's not enough slots to accomodate them all?
+
             // 4.
-            // Sort by dApp ID, in ascending order (unstable sort should be faster, and stability is guaranteed due to lack of duplicated Ids).
-            // TODO & Idea: perhaps use BTreeMap instead? It will "sort" automatically based on dApp Id, and we can efficiently remove entries,
-            // reducing PoV size step by step.
-            // It's a trade-off between speed and PoV size. Although both are quite minor, so maybe it doesn't matter that much.
+            // Sort by dApp ID, in ascending order (unstable sort should be faster, and stability is "guaranteed" due to lack of duplicated Ids).
             dapp_tiers.sort_unstable_by(|first, second| first.dapp_id.cmp(&second.dapp_id));
 
             // 5. Calculate rewards.
             let tier_rewards = tier_config
```

Comment:
Having a formula as comment for the reward calculation algo would be helpful.

Reply:
Added!